myapp/pps_regressor.py

```diff
@@ -0,0 +1,31 @@
+import numpy as np
+import pandas as pd 
+import lightgbm as lgb
+import json
+
+DENSITIES = {0:"SINGLE FAM", 1: "PLEX", 2: "CONDO", 3: "HRISE CONDO"}
+SEGMENTS = {0: 'D', 1: 'C', 2: 'B'}
+
+def prediction(input_dict):
+    model = lgb.Booster(model_file='noyear.txt')
+    lst = [(input_dict)]
+    tester = pd.DataFrame(lst)
+    print(tester)
+    tester['Log-Built'] = np.log(tester['Year Built'])
+    tester['Log-SqFt'] = np.log(int(tester['Square Footage']))
+    tester['Neighborhood'] = tester['Neighborhood'].astype(float)
+    with open('meanpps.json', 'r') as nbhdpps:
+        nbhd_dict = json.load(nbhdpps)
+    meanpps = pd.DataFrame(nbhd_dict, index=None)
+    meanpps.index=meanpps.index.astype(float)
+    nbhds = list(meanpps.index)
+    tester=pd.merge(how = 'left', left = tester, right = meanpps,left_on="Neighborhood", right_index=True)
+    tester['Log-NbhdPPS'] = np.log(tester['NeighborhoodPPS'])
+    tester['Neighborhood'] = str(tester['Neighborhood'])
+    tester['Building-Grade'] = tester['Building-Grade'].map(lambda x: SEGMENTS[x])
+    tester['Building-Type-Custom'] = tester['Building-Type-Custom'].map(lambda x: DENSITIES[x])
+    X = pd.get_dummies(tester[['Year', 'Log-Built', 'Neighborhood',  'Building-Type-Custom', 
```

Comment:
The main problem here is that it will be getting the dummies for this 1 row tests data frame. However, because tester only has one entry, the dummies will be only whats available in the one row. However, your model has columns for every single possible Grade available in the training data. You need to one-hot encode in such a way as to get the same columns as in you training set.